p2p/peer.go

```diff
@@ -139,3 +140,26 @@ func (p *MutablePeer) Subscribe(sub func(Peer)) {
 
 	p.subs = append(p.subs, sub)
 }
+
+// VerifyP2PKey returns an error if the p2pkey doesn't match any lock operator ENR.
+func VerifyP2PKey(peers []Peer, key *ecdsa.PrivateKey) error {
+	wantBytes := crypto.CompressPubkey(&key.PublicKey)
+
+	for _, p := range peers {
+		pk, err := p.ID.ExtractPublicKey()
+		if err != nil {
+			return errors.Wrap(err, "extract pubkey from peer id")
+		}
+
+		gotBytes, err := pk.Raw()
+		if err != nil {
+			return errors.Wrap(err, "key to bytes")
+		}
+
+		if bytes.Equal(wantBytes, gotBytes) {
+			return nil
+		}
+	}
+
+	return errors.New("private key not matching any operator ENR") // This message seems to know something about the context in which it is used :(
```

Comment:
i didn't understand the comment ;)

Reply:
How about: `unknown private key provided, it doesn't match any public key encoded inside the operator ENRs`

Reply:
This functions isn't provided with any operators or ENRs so the fact that it mentions it in the error message is assuming specific implicit external use-case